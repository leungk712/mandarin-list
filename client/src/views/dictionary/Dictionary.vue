<template>
  <div>
    <v-container class="container-border dictionary-container mb-6">
      <p class="display-1 my-4">Chinese Dictionary</p>
      <v-divider />
      <p class="my-4 title">
        {{ dictionaryText }}
      </p>
      <v-row justify="center" class="my-6">
        <v-btn
          data-testid="dictionary-link-btn"
          class="dictionary-link-btn"
          color="blue lighten-1 white--text"
          large
        >
          <v-icon class="mr-2">open_in_new</v-icon>
          <a
            href="https://mandarin-list-dictionary.netlify.com"
            target="_blank"
            class="text-decoration-none white--text"
          >
          Dictionary
          </a>
        </v-btn>
      </v-row>
      <v-img src="@/assets/misc/mandarin-list-dictionary.png" />
    </v-container>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from "vue-property-decorator";
import { dictionaryText } from "@/helpers/dictionary";

@Component({
  name: "Dictionary"
})
export default class Dictionary extends Vue {
  // ===== Data ===== //
  public dictionaryText = dictionaryText;

  // ===== Methods ===== //

  // ===== Computed ===== //

  // ===== Lifecycle Hooks ===== //
}
</script>

<style scoped>
  .dictionary-container {
    width: 80%;
  }
</style>
