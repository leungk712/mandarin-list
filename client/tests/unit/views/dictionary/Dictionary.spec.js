import { createLocalVue, shallowMount } from "@vue/test-utils";
import Dictionary from "@/views/dictionary/Dictionary.vue";
import Vue from "vue";
import Vuex from "vuex";
import Vuetify from "vuetify";

Vue.use(Vuetify);

const localVue = createLocalVue();
localVue.use(Vuex);

const elem = document.createElement("div");
if (document.body) {
    document.body.appendChild(elem);
};

document.body.setAttribute("data-app", true);

const actions = {};

const mockStore = {};

describe('Dictionary', () => {
    let wrapper;
    const store = new Vuex.Store({
        modules: mockStore
    });
    const mountFunction = (options, store = mockStore) => {
        return shallowMount(Dictionary, {
           attachTo: elem,
           localVue,
           store,
           vuetify: new Vuetify(),
           ...options
        });
    };
    describe("template", () => {
        it("should match snapshot", () => {
            wrapper = mountFunction({}, store);
            expect(wrapper.vm.$el).toMatchSnapshot();
            wrapper.destroy();
        });
    });
});