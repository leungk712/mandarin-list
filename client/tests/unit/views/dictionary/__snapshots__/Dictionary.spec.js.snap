// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dictionary template should match snapshot 1`] = `
<div>
  <v-container-stub
    class="container-border dictionary-container mb-6"
    tag="div"
  >
    <p
      class="display-1 my-4"
    >
      Chinese Dictionary
    </p>
     
    <v-divider-stub />
     
    <p
      class="my-4 title"
    >
      
      Due to the dictionary package utilized being too large, resulting in decreased application performance,
the dictionary is on a separate website which you can click below to access. Please be patient with the loading time...
    
    </p>
     
    <v-row-stub
      class="my-6"
      justify="center"
      tag="div"
    >
      <v-btn-stub
        activeclass=""
        class="dictionary-link-btn"
        color="blue lighten-1 white--text"
        data-testid="dictionary-link-btn"
        large="true"
        tag="button"
        type="button"
      >
        <v-icon-stub
          class="mr-2"
        >
          open_in_new
        </v-icon-stub>
         
        <a
          class="text-decoration-none white--text"
          href="https://mandarin-list-dictionary.netlify.com"
          target="_blank"
        >
          
        Dictionary
        
        </a>
      </v-btn-stub>
    </v-row-stub>
     
    <v-img-stub
      options="[object Object]"
      position="center center"
      src="@/assets/misc/mandarin-list-dictionary.png"
      transition="fade-transition"
    />
  </v-container-stub>
</div>
`;
